### PR TITLE
Workbench: update to electron v25, jest v29

### LINKED ISSUE #1223: Do major node package upgrades for Workbench
Lots are outdated: `yarn outdated`

It would be nice to upgrade, but lots of breaking changes are possible, so I think we should do this after the `3.13.0` release.

### PR DESCRIPTION
**Pull request by davemfish**
The main thing here is upgrading to electron 25. Most of the changes were necessitated by breaking changes in v20, related to even more restrictions on using `node` packages in the preload script. Specifically, we no longer import `electron-log` in preload or renderer. Instead we do all logging from the renderer via IPC to main. 

At the same time, something broke in the `babel` transformations we were using for the preload module. I decided to move away from `babel` and use `vite` instead, especially since the preload script is now even more targeted to the browser than in the past. We could consider doing the same change for `main` module code, but so far that still just works. 

Some relatively minor changes related to the jest upgrade. `mockRestore` behavior somehow changed. And they _finally_ exposed an option to print out all the failing tests at the end of the summary.

I also scoped updating React to v18, but decided to that separately (see #1325 ). 

Fixes #1223 

## Checklist
- [ ] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

### CODE REVIEW

File: workbench/src/main/main.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ export const createWindow = async () => {
   setupRendererLogger();
   await getFlaskIsReady();
 
-  const devModeArg = ELECTRON_DEV_MODE ? '--devMode' : '';
+  const devModeArg = ELECTRON_DEV_MODE ? '--devmode' : '';
```
Comment (davemfish) [inline workbench/src/main/main.js:93]: this was silently getting lowercased anyway, which was messing with our args parsing in `preload/api.js`

File: workbench/src/preload/api.js [hunk 2]
```diff
@@ -18,17 +18,17 @@ const ipcRendererChannels = [
...
 // As well as being loaded by the electron preload script,
 // this module is loaded by the jest setup script in order to
 // mock APIs that are normally setup during electron preload.
 // In the jest case, we do not have userPaths data.
 const userguidePath = (!isDevMode && userPaths)
-  ? `file:///${userPaths.resourcesPath}/documentation`
+  ? `file:///${userPaths.resourcesPath}/documentation` // as a URL
   : ''; // In DevMode, local UG is served at the root path
 const electronLogPath = (userPaths)
-  ? `${userPaths.userData}/logs/main.log`
```
Comment (davemfish) [inline workbench/src/preload/api.js]: These forward slashes made for an invalid path on Windows when mixed with backslashes in `userPaths.userData`. So I moved this whole construction over to the main process where we can use `path.join`.

The userguide path, which is actually read by a browser as a URL, works fine this way. And that URL is tested by puppeteer, so definitely works in dev & production on all OS.

**Comment by davemfish** [comment]
@emlys I made those few changes we discussed, and also created #1333.

FYI, if you run these changes you might notice there are even more warnings on startup emanating from the react-devtools-extension. So I re-opened #862 .

**Review by emlys — APPROVED** [review]
looks good!

**Pull request merged**